google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubTest.java

```diff
@@ -291,6 +292,26 @@ public void testFeatureFlags() throws InterruptedException, IOException, Executi
     assertThat(featureFlags.getLastScannedRowResponses()).isTrue();
   }
 
+  @Test
+  public void testPingAndWarmFeatureFlags()
+      throws InterruptedException, IOException, ExecutionException {
+    EnhancedBigtableStubSettings settings =
+        defaultSettings.toBuilder().setRefreshingChannel(true).build();
+    try (EnhancedBigtableStub ignored = EnhancedBigtableStub.create(settings)) {
+      Preconditions.checkState(
+          !fakeDataService.pingRequests.isEmpty(), "Ping request was not sent during setup");
+      Metadata metadata = metadataInterceptor.headers.take();
+
+      String encodedFeatureFlags =
+          metadata.get(Key.of("bigtable-features", Metadata.ASCII_STRING_MARSHALLER));
+      FeatureFlags featureFlags =
+          FeatureFlags.parseFrom(BaseEncoding.base64Url().decode(encodedFeatureFlags));
+
+      assertThat(featureFlags.getReverseScans()).isTrue();
+      assertThat(featureFlags.getLastScannedRowResponses()).isTrue();
```

Comment:
routingCookie, retryInfo, clientSideMetricsEnabled should also be true

Reply:
done